Documentation/Diagnostics/PH2140.md

```diff
@@ -0,0 +1,47 @@
+# PH2140: Prefer readonly parameters
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2140 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [PreferReadOnlyParametersAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/PreferReadOnlyParametersAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled by default | Yes |
+
+## Introduction
+
+Methods that do not modify a parameter should declare that parameter as read-only. This enabled calling code to use read-only types, which increase readability of the code. Please note this is similar to the `const` keyword in C++.
+
+In C# read-only types are availoable for Collections and the `Span` class. 
+
+## How to solve
+
+When not modifying the collection, declare the parameter as read-only, using for example `IReadOnlyList<>` iso `List<>`.
+
+## Example
```

Comment:
Also note that many times simply IEnumerable is sufficient (i.e., not even IReadOnly). It would be cool to differentiate that (e.g., in a CodeFixer)